### PR TITLE
delete instance

### LINKED ISSUE #86: Delete doesn't clean up instance
Delete method returns with 200 OK, but the instance status is still deleting and some resources are still in kubernetes. No trace of error in logs.

### PR DESCRIPTION
**Pull request by macat**
- Deletes instance from etcd after deleting kubernetes resources

Fixes #86

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -261,10 +261,16 @@ func del(s *Server, pID string, ID string) (*instance.Instance, error) {
 
 	ds := services.NewDeploymentService(s.store, s.playbooks, s.manifests)
 
-	err = ds.DeleteAndNotify(i)
-	if err != nil {
+	if err := ds.DeleteAndNotify(i); err != nil {
```
Comment (therealplato) [inline server/server.go]: why are both `ds.DeleteAndNotify(i)` and `c.is.Delete(i)` being called? what do they do differently?

Comment (macat) [inline server/server.go]: `ds.DeleteAndNotify` will delete kubernetes resources while `c.is.Delete` deletes the instance from etcd.

**Comment by cored** [comment]
@macat Are there tests for this? apart from that :+1:

**Comment by macat** [comment]
@cored added check to make sure we are now deleting the instance.

**Comment by poopoothegorilla** [comment]
+1

**Pull request merged**